### PR TITLE
2689: artifacts are content

### PR DESCRIPTION
**Pull request by raman-bt**

### CODE REVIEW

File: bindings/java/src/org/sleuthkit/datamodel/BlackboardAttribute.java [hunk 1]
```diff
@@ -518,8 +521,32 @@ public String getDisplayString() {
...
+					} 
+				} catch (TskException ex) {
+					// contnue with TimeZone.getDefault();
+				}
+				
+				String timeString = "0000-00-00 00:00:00";
+				long epochSeconds = getValueLong();
+				if (epochSeconds != 0) {
+					synchronized (DATE_FORMATTER) {
+						DATE_FORMATTER.setTimeZone(tzone);
```
Comment (bcarrier) [inline bindings/java/src/org/sleuthkit/datamodel/BlackboardAttribute.java]: I'd rather that this go into a new AbstractFile.epochToTime method that takes in a TimeZone argument.   The code in the attribute tries to find the Timezone and then calls either the new epochToTime() with that TimeZone or calls the original one. 

Also, epochToTime() is in AbstractFile, but all of the references are to FsContent. We should update those to make that more obvious. It took me a bit to find it.

Comment (raman-bt) [inline bindings/java/src/org/sleuthkit/datamodel/BlackboardAttribute.java]: Might there be another home for epochToTime() functions?

 AbstractFile isn't the the most intuitive p[lace I'd go looking for time conversion utility methods.

File: bindings/java/src/org/sleuthkit/datamodel/BlackboardArtifact.java [hunk 2]
```diff
@@ -316,7 +561,7 @@ public int hashCode() {
 	 */
 	@Override
 	public String toString() {
-		return "BlackboardArtifact{" + "artifactID=" + artifactId + ", objID=" + objId + ", artifactTypeID=" + artifactTypeId + ", artifactTypeName=" + artifactTypeName + ", displayName=" + displayName + ", Case=" + sleuthkitCase + '}'; //NON-NLS
+		return "BlackboardArtifact{" + "artifactID=" + artifactId + ", objID=" + getObjectID() + ", artifactObjID=" + artifactObjId + ", artifactTypeID=" + artifactTypeId + ", artifactTypeName=" + artifactTypeName + ", displayName=" + displayName + ", Case=" + getSleuthkitCase() + '}'; //NON-NLS
```
Comment (raman-bt) [inline bindings/java/src/org/sleuthkit/datamodel/BlackboardArtifact.java]: This isn't applicable any more if we BlackboardArtifact isn't extending AbstractContent. Right?

File: bindings/java/src/org/sleuthkit/datamodel/BlackboardAttribute.java [hunk 3]
```diff
@@ -518,8 +521,23 @@ public String getDisplayString() {
...
+					final Content dataSource = getParentArtifact().getDataSource();
+					if ((dataSource != null) && (dataSource instanceof Image )) {
+						// return the date/time string in the timezone associated with the datasource,
+						Image  image = (Image) dataSource;
+						TimeZone tzone = TimeZone.getTimeZone(image.getTimeZone());
+						return epochToTime(getValueLong(), tzone);
+					} 
+				} catch (TskException ex) {
+					// contnue with TimeZone.getDefault();	
+				}
```
Comment (raman-bt) [inline bindings/java/src/org/sleuthkit/datamodel/BlackboardAttribute.java]: I agree with this. But this is in Sleuthkit datamodel and I didn't find any logging in those modules. I didn't think this warranted throwing an exception. Lets discuss alternatives.

File: bindings/java/src/org/sleuthkit/datamodel/BlackboardArtifact.java (hunk 2 above)
Comment (rcordovano) [inline bindings/java/src/org/sleuthkit/datamodel/BlackboardArtifact.java]: Quite true, sorry, I will delete this comment.

**Comment by rcordovano** [comment]
Closing, because we are going to bring this into the artifacts-are-content branch and do an integration into develop later.

**Pull request closed without merging**